### PR TITLE
useEndpoint Upates

### PR DESCRIPTION
**Pull request by ksingh4096**
### 1. Please briefly explain what it is you coded

-  useEndpoint: now has the ability to make an undoable request
-  useSnack: snacks have been updated to meet accessibility standards (single variant) 

### 2. Please briefly explain your approach (new components added? modified? removed?)

- useEndpoint modified with undoableRequest function
- Added a new design doc for useEndpoint  
- modified any snack calls updated to new syntax (removal of variant parameter)
- useSnack updated to typescript

### 3. Any packages added/updated/removed
- No
### (Optional) 4. Any feature recommendations/insights/questions/comments/etc.?
- Potential future changes to be made to useEndpoint using a cleanup function in React.useEffect()

### CODE REVIEW

File: src/hooks/useEndpoint.ts [hunk 1]
```diff
@@ -76,11 +79,56 @@ export default function useEndpoint<T>(
...
+            snack(undo?.message || 'Request Sent', {
+                action: React.createElement( 
+                    Button,
+                    {
+                        onClick: () => {
+                            if (isMounted === true) {
+                                _onUndo();
+                                isStopped = true;
+                                setIsLoading(false);
+                                window.removeEventListener('beforeunload', closeFunction, false);
```
Comment (DevelopmentByDavid) [inline src/hooks/useEndpoint.ts:104]: So from the code we always add the event listener, but then conditionally check if the component is mounted to remove it, correct? I think we should probably remove it regardless of whether or not the component is mounted.  Most of the time I'm checking if the component is mounted so I don't use a `setState` after unmount, but this is a `window.something`, which is global.

**Review by DevelopmentByDavid — COMMENTED** [review]
Looks great! Minor comments.

File: src/App.tsx [hunk 2]
```diff
@@ -26,10 +26,17 @@ export default function App() {
...
+                <SnackContext maxSnack={1}>
+                    <BrowserRouter>
+                        <Pages />
+                    </BrowserRouter>
+=======
                 <SnackContext maxSnack={3}>
                     <UserContext.Provider value={{ user, setUser }}>
                         <Routes />
                     </UserContext.Provider>
+>>>>>>> upstream/master
```
[the remark this answers is not in the document]
Comment (ksingh4096) [reply src/App.tsx]: fixed

File: src/hooks/useEndpoint.ts (hunk 1 above)
Comment (ksingh4096) [reply src/hooks/useEndpoint.ts]: I moved the removeEventListener call before the isMounted check in onClick. It is also called as the first thing in onExited.

**Review by DevelopmentByDavid — APPROVED** [review]

**Pull request merged**